internal/logs/workload.go

```diff
@@ -0,0 +1,199 @@
+package logs
+
+import (
+	"context"
+	"time"
+
+	"git.sr.ht/~spc/go-log"
+	"github.com/project-flotta/flotta-device-worker/internal/workload"
+	"github.com/project-flotta/flotta-device-worker/internal/workload/podman"
+	"github.com/project-flotta/flotta-operator/models"
+)
+
+type transportLog struct {
+	transport transport
+	log       *FifoLog
+	cancel    context.CancelFunc
+}
+
+func (t *transportLog) Start() {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.cancel = cancel
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				entry, err := t.log.ReadLine()
+				if err != nil {
+					log.Error("Error on read log info", err)
+				}
+				if entry == nil {
+					time.Sleep(elapsedTime)
+				} else {
+					err = t.transport.WriTeLog(entry)
+					if err != nil {
+						// cannot send, pushed again to the queue.
+						err := t.log.Write(entry)
+						if err != nil {
+							log.Error("cannot return log entry to the buffer log after transport failed", err)
```

Comment:
In that case that log entry is lost. In case of communication problems of the device, when transport can't connect for long time we might lose big chunks of logs. Is there a way for dealing with that I don't see?

Reply:
we pushed again to the queue, and it has a buffer. Maybe in future we can add a fifolog/failed entry to a file, and sync when connectivity is off. This is the best effort and buffers size.

Reply:
But the buffer has limited size and can be full very quickly: https://github.com/project-flotta/flotta-device-worker/pull/101/files#diff-b642ccca5637f0ce9e7df1edc0d12111f67b1604b13bb89edc63830ba147ab9bR76.

Reply:
We can extend in future, users have the way to define how big the buffer can be, as defined in the doc. I'm happy to move to file, but what happens if the device is running without internet for a month and the log is duplicated and at the end the disk is full, it's a bit of chicken-egg problem, and at the end, it'll lose log files. 

The source of truth is always the device, what we proposed here is a best-effort solution IMHO.

Reply:
We can pause reading from the log when the buffer is full until something is eventually sent successfully. In case of reboot while disconnected it makes no difference if the source logs are unprocessed or lost from memory. To prevent that, we might somehow preserve information about the last read and sent line - to restart reading log file from that place after restart.

Reply:
In that case, you're going to lose the logs that are going to be rotated on the journald process(Roy is working on that), so at the end is a similar situation that you will lose logs.